extensions/blocks/podcast-player/podcast-player.js

```diff
@@ -0,0 +1,372 @@
+		}
+
+		this.setPlayerState( STATE_PAUSED );
+	}
+
+	/**
+	 * Play current audio.
+	 * @public
+	 */
+	play() {
+		// Ignoring exceptions as they are handled globally from the audio element.
+		this.audio.play().catch( noop );
+	}
+
+	/**
+	 * Pause current audio.
+	 * @public
+	 */
+	pause() {
+		this.audio.pause();
+	}
+
+	/**
+	 * Stop current audio, clean selected track.
+	 * @public
+	 */
+	stop() {
+		this.pause();
+		this.setTrackState( this.currentTrack, false );
+		this.currentTrack = -1;
+	}
+
+	/**
+	 * Toggle playing state.
+	 * @public
+	 */
+	togglePlayPause() {
+		if ( this.audio.paused ) {
+			this.play();
+		} else {
+			this.pause();
+		}
+	}
+
+	/**
+	 * Select a track and play/pause, as needed.
+	 * @public
+	 * @param {number} track The track number
+	 */
+	selectTrack( track ) {
+		// Current track already selected.
+		if ( this.currentTrack === track ) {
+			this.togglePlayPause();
+			return;
+		}
+
+		// Something else is playing.
+		if ( this.currentTrack !== -1 ) {
+			this.stop();
+		}
+
+		// Load a new track.
+		this.loadAndPlay( track );
+	}
+
+	/**
+	 * Set a player state classname.
+	 * @private
+	 * @param {string} state Player state.
+	 */
+	setPlayerState( state ) {
+		this.playerState = state;
+		this.block.classList.remove( STATE_ERROR, STATE_PAUSED, STATE_PLAYING );
+		this.block.classList.add( state );
+	}
+
+	/**
+	 * Load audio from the track, start playing.
+	 * @private
+	 * @param {number} track The track number
+	 */
+	loadAndPlay( track ) {
+		const trackData = this.getTrack( track );
+		if ( ! trackData ) {
+			return;
+		}
+		this.audio.src = trackData.src;
+		this.currentTrack = track;
+		this.play();
+		this.setTrackState( track, true );
+	}
+
+	/**
+	 * Visually mark track active or inactive.
+	 * @private
+	 * @param {number} track The track number
+	 * @param {boolean} isActive True if track is active
+	 * @param {boolean} isError True if track couldn't be played
+	 */
+	setTrackState( track, isActive, isError = false ) {
+		// Mark track element as active/inactive.
+		const el = this.trackElements[ track ];
+
+		// Active state classname.
+		if ( isActive ) {
+			el.classList.add( 'is-active' );
+		} else {
+			el.classList.remove( 'is-active' );
+		}
+
+		// Maintain pressed state.
+		el.querySelector( '[aria-pressed]' ).setAttribute( 'aria-pressed', JSON.stringify( isActive ) );
+
+		// Add or remove error.
+		const errorEl = el.querySelector( '.podcast-player__episode-error' );
+		if ( isError && ! errorEl ) {
+			const trackData = this.getTrack( track );
+			el.appendChild( this.renderErrorWithLink( trackData && trackData.link ) );
+		} else if ( ! isError && errorEl ) {
+			errorEl.remove();
+		}
+
+		const iconEl = el.querySelector( '.podcast-player__episode-status-icon' );
+		if ( iconEl ) {
+			if ( isActive ) {
+				if ( isError ) {
+					iconEl.innerHTML = errorIcon;
+				} else {
+					iconEl.innerHTML = soundIcon;
+				}
+			} else {
+				iconEl.innerHTML = '';
+			}
+		}
```

Comment:
Any good way to reduce the nesting here?

Not well thought out but something like...
```
if ( ! iconEl ) {
  return;
} else if( ! isActive ) {
  iconEl.innerHTML = '';
} else if ( isError ) {
  iconEl.innerHTML = errorIcon;
} else {
  iconEl.innerHTML = soundIcon;
}
```

It's one level, but maybe it's actually harder to mentally parse this with all the `!` negations?

Reply:
I'm also not a fan of nested conditions but I find the linear one harder to understand to be honest. Also, an early return might be surprising in a method that's not single purpose and if we were to add more functionality at the bottom of the function, it might not be immediately clear why it's sometimes not being called.

Reply:
Agreed that there are issues with the single column one. If there's a way to reduce it to even two nests, that would be nice. That said, at least your deep nesting is easy enough to understand :)